clusterMembership/clusterMembership.go

```diff
@@ -0,0 +1,21 @@
+package clusterMembership
+
+
+type Node interface {
+	GetId() string
+	SendMessage(msg string) error
+}
+
+type Cluster interface {
+	/*
+	 * Returns a Slice of Node Ids in the Cluster
+	 */
+	GetMembers() []string
```

Comment:
The idea here is we don't want users directly invoking the SendMessage method on the nodes.

There is a race condition where I give a process the list of nodes, and then for a dynamic cluster a node is removed.  We want SendMessage to go through the Cluster interface.  So if the message is trying to be sent to a node that no longer exists, we return an error.  

This may need to be modified if we want to implement work distributions that depend on Node specific information like load etc.. that the application state machine can't keep track of itself.